src/System.Windows.Extensions/tests/System/Drawing/Printing/MarginsConverterTests.cs

```diff
@@ -0,0 +1,160 @@
+// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using System.Collections;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.ComponentModel.Design.Serialization;
+using System.Globalization;
+using Xunit;
+
+namespace System.Drawing.Printing.Tests
+{
+    public class MarginsConverterTests
+    {
+        [Fact]
+        public void CanConvertFrom()
+        {
+            MarginsConverter mc = new MarginsConverter();
+
+            // try once with then once without context
+            for (var context = new MyTypeDescriptorContext(); context != null; context = null)
+            {
+                Assert.True(mc.CanConvertFrom(context, typeof(string)));
+                Assert.False(mc.CanConvertFrom(context, typeof(Guid)));
+                Assert.False(mc.CanConvertFrom(context, typeof(object)));
+                Assert.False(mc.CanConvertFrom(context, typeof(int)));
+            }
+        }
+
+        [Fact]
+        public void CanConvertTo()
+        {
+            MarginsConverter mc = new MarginsConverter();
+
+            // try once with then once without context
+            for (var context = new MyTypeDescriptorContext(); context != null; context = null)
+            {
+                Assert.True(mc.CanConvertTo(context, typeof(string)));
+                Assert.False(mc.CanConvertTo(context, typeof(Guid)));
+                Assert.False(mc.CanConvertTo(context, typeof(object)));
+                Assert.False(mc.CanConvertTo(context, typeof(int)));
+            }
+        }
+
+        [Fact]
+        public void CreateInstance()
+        {
+            MarginsConverter mc = new MarginsConverter();
+            MyTypeDescriptorContext context = new MyTypeDescriptorContext();
+
+            IDictionary values = new Dictionary<string, string>();
+            values.Add("Left", "1");
+            values.Add("Right", "2");
+            values.Add("Top", "3");
+            Assert.Throws<ArgumentException>(() => mc.CreateInstance(context, values));
+            values.Add("Bottom", "4");
+            if (PlatformDetection.IsFullFramework)
+            {
+                Assert.Throws<ArgumentException>(() => mc.CreateInstance(context, values));
```

Comment:
I'm keeping this because I believe FullFramework is behaving incorrectly:
https://referencesource.microsoft.com/#System.Drawing/commonui/System/Drawing/Printing/MarginsConverter.cs,167

So for CreateInstance I am using the mono version.

Reply:
What is incorrect about the FullFramework behavior? I understand that the Mono and FullFramework behaviors are different; but I do not see why the Mono behavior is the right one.

Reply:
In the mono code, we could also create instance if the property value objects were able to be cast to int not just be int.

You are right the code is not incorrect, but mono code was more lenient.
I updated the test case instead.